python/sdss_access/path/path.py

```diff
@@ -1529,34 +1557,200 @@ def pad_fieldid(self, filetype, **kwargs):
+    def spcoaddfolder(self, filetype, **kwargs):
+        ''' Returns the reorganized subfolder structure for the BOSS idlspec2d run2d version
 
         Parameters
-        ----------
+        ---------
         filetype : str
-            File type parameter.
-        tileid : int or str
-            LVM Tile ID number.  Will be converted to int internally.
+            File type parameter
+        run2d : str
+            BOSS idlspec2d run2d version
+        coadd : str
+            Name of the custom coadd schema
+        Returns
+        -------
+        sptypefolder : str
+        '''
+
+        run2d = kwargs.get('run2d', None)
+        coaddname = kwargs.get('coadd', None)
+        
+        if (not run2d):
+            return ''
+        elif (('v5' in run2d) or (str(run2d) in ['26','103','104']) or
+              ('v6_0' in run2d) or ('v6_1' in run2d)):
+            return ''
+        else:
+            if filetype.lower() in ['spall_coadd','spall-lite_coadd','spallline_coadd']:
+                return('summary')
+            elif filetype.lower() in ['speclite_coadd','specfull_coadd',
+                                     'spallfield_coadd','spalllinefield_coadd']:
+                return(coaddname)
+            else:
+                return('fields')
 
+
+    def spcoaddgrp(self, filetype, **kwargs):
+        ''' Returns the coadd group (field group analog) subfolder structure for the BOSS idlspec2d run2d version
+
+        Parameters
+        ---------
+        filetype : str
+            File type parameter
+        run2d : str
+            BOSS idlspec2d run2d version
+        coadd : str
+            Name of the custom coadd schema
         Returns
         -------
-        tileidgrp : str
-            Tile ID group directory in the format ``NNNNXX``.
+        spcoaddgrp : str
+        '''
+        run2d = kwargs.get('run2d', None)
+        coaddname = kwargs.get('coadd', None)
+
+        if (not run2d):
+            return ''
+        elif (('v5' in run2d) or (str(run2d) in ['26','103','104']) or
+              ('v6_0' in run2d) or ('v6_1' in run2d)):
+            return ''
+        else:
+            return(coaddname)
+            
+    def sptypefolder(self, filetype, **kwargs):
+        ''' Returns the reorganized subfolder structure for the BOSS idlspec2d run2d version
 
+        Parameters
+        ---------
+        filetype : str
+            File type parameter
+        run2d : str
+            BOSS idlspec2d run2d version
+        Returns
+        -------
+        sptypefolder : str
         '''
 
-        tileid = kwargs.get('tileid', None)
-        if not tileid:
-            return '0000XX'
-        elif '*' in str(tileid):
-            return '{0}XX'.format(tileid)
-        return '{:0>4d}XX'.format(int(tileid) // 1000)
+        run2d = kwargs.get('run2d', None)
+        
+        if (not run2d):
+            return ''
+        elif ('v5' in run2d) or (str(run2d) in ['26','103','104']):
+            return ''
```

Comment:
you can combine these into one conditional

Reply:
Fixed this and other similar instances.